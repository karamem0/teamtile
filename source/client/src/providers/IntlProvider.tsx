//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import {
  IntlProvider as Provider,
  createIntl,
  MessageFormatElement
} from 'react-intl';

import en from '../translations/compiled/en.json';
import ja from '../translations/compiled/ja.json';

const translations: { [key: string]: Record<string, MessageFormatElement[]> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language.substring(0, 2)]
});

function IntlProvider(props: React.PropsWithChildren<unknown>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
