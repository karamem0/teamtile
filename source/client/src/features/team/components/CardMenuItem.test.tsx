//
// Copyright (c) 2021-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import IntlProvider from '../../../providers/IntlProvider';
import Presenter from './CardMenuItem.presenter';
import { render } from '@testing-library/react';

it('should create a shapshot', () => {
  const params = {
    icon: (
      <div data-testid="test-Icon" />
    ),
    title: 'Title'
  };
  const { asFragment } = render(
    <IntlProvider>
      <Presenter {...params} />
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
