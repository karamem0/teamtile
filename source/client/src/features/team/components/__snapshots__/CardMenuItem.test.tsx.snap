// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: auto auto;
  grid-gap: 0.25rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  cursor: pointer;
}

<span
    aria-label="Title"
    class="fui-Text emotion-0 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
  >
    <button
      class="fui-Button r1alrhcs ___cpe5h10_1yhfyj7 f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f ft1hn21 fuxngvv fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1dfjoow f1j98vj9 f1tme0vf f4xjyn1 f18onu3q f9ddjv3 f18ktai2 fwbmr0d f44c6la"
      type="button"
    >
      <span
        class="fui-Button__icon rywnvv2"
      >
        <div
          data-testid="test-Icon"
        />
      </span>
    </button>
  </span>
</DocumentFragment>
`;
